cmd/notary/keys.go

```diff
@@ -4,104 +4,63 @@ import (
+func init() {
+	cmdKeys.AddCommand(cmdKeysRemoveRootKey)
+	cmdKeys.AddCommand(cmdKeysGenerateRootKey)
+}
 
 var cmdKeys = &cobra.Command{
 	Use:   "keys",
-	Short: "Operates on keys.",
-	Long:  "operations on signature keys and trusted certificate authorities.",
+	Short: "Operates on root keys.",
+	Long:  "operations on private root keys.",
 	Run:   keysList,
 }
 
-func init() {
-	cmdKeys.AddCommand(cmdKeysTrust)
-	cmdKeys.AddCommand(cmdKeysRemove)
-	cmdKeys.AddCommand(cmdKeysGenerate)
-}
-
-var cmdKeysRemove = &cobra.Command{
-	Use:   "remove [ Subject Key ID ]",
-	Short: "Removes trust from a specific certificate authority or certificate.",
-	Long:  "remove trust from a specific certificate authority.",
-	Run:   keysRemove,
+var cmdKeysRemoveRootKey = &cobra.Command{
+	Use:   "remove [ keyID ]",
+	Short: "Removes the root key with the given keyID.",
+	Long:  "remove the root key with the given keyID from the local host.",
+	Run:   keysRemoveRootKey,
 }
 
-var cmdKeysTrust = &cobra.Command{
-	Use:   "trust [ certificate ]",
-	Short: "Trusts a new certificate.",
-	Long:  "adds a the certificate to the trusted certificate authority list.",
-	Run:   keysTrust,
+var cmdKeysGenerateRootKey = &cobra.Command{
+	Use:   "generate [ algorithm ]",
+	Short: "Generates a new root key with a given algorithm.",
+	Long:  "generates a new root key with a given algorithm.",
+	Run:   keysGenerateRootKey,
 }
 
-var cmdKeysGenerate = &cobra.Command{
-	Use:   "generate [ GUN ]",
-	Short: "Generates a new key for a specific GUN.",
-	Long:  "generates a new key for a specific Global Unique Name.",
-	Run:   keysGenerate,
-}
-
-// keysRemove deletes Certificates based on hash and Private Keys
-// based on GUNs.
-func keysRemove(cmd *cobra.Command, args []string) {
+// keysRemoveRootKey deletes a root private key based on ID
+func keysRemoveRootKey(cmd *cobra.Command, args []string) {
 	if len(args) < 1 {
 		cmd.Usage()
-		fatalf("must specify a SHA256 SubjectKeyID of the certificate")
-	}
-
-	//TODO (diogo): Validate Global Unique Name. We probably want to reject 1 char GUNs.
-	gunOrID := args[0]
-
-	// Try to retrieve the ID from the CA store.
-	cert, err := caStore.GetCertificateByCertID(gunOrID)
-	if err == nil {
-		fmt.Printf("Removing: ")
-		printCert(cert)
-
-		// If the ID is found, remove it.
-		err = caStore.RemoveCert(cert)
-		if err != nil {
-			fatalf("failed to remove certificate from KeyStore")
-		}
-		return
+		fatalf("must specify the key ID of the root key to remove")
 	}
 
-	// Try to retrieve the ID from the Certificate store.
-	cert, err = certificateStore.GetCertificateByCertID(gunOrID)
-	if err == nil {
-		fmt.Printf("Removing: ")
-		printCert(cert)
-
-		// If the ID is found, remove it.
-		err = certificateStore.RemoveCert(cert)
-		if err != nil {
-			fatalf("failed to remove certificate from KeyStore")
-		}
-		return
+	keyID := args[0]
+	if len(keyID) != 64 {
+		fatalf("please enter a valid root key ID")
 	}
+	parseConfig()
```

Comment:
Would it make sense to call parseConfig() in main so we don't have to call it in every command function?